crates/hir/src/lib.rs

```diff
@@ -1967,12 +1967,18 @@ impl Type {
         let env = self.ty.environment.clone();
         let krate = krate.id;
 
+        let from_module = match self.as_adt() {
+            Some(adt) => Some(adt.module(db).id),
+            None => None,
+        };
+
         method_resolution::iterate_method_candidates(
             &canonical,
             db,
             env,
             krate,
             traits_in_scope,
+            from_module,
```

Comment:
Not sure if this is correct.

Reply:
Hmm actually I think this would need to be passed by the caller. This is used by completion, and this way it'd always complete from the visibility of the self type, instead of wherever we're typing. We should pass a module instead of the `krate`. It'd be OK to punt on that in this PR though, and just pass `None` for now.